docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,727 @@
+  # The reason is that the DNS management is not part of the Gardener core code base but externalized, hence, it might use other
+  # key names than Gardener itself.
+  # The actual values here depend on the DNS extension that is installed to your landscape.
+  # For example, check out https://github.com/gardener/external-dns-management and find a lot of example secret manifests here:
+  # https://github.com/gardener/external-dns-management/tree/master/examples
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: SecretBinding
+metadata:
+  name: secretbinding1
+  namespace: garden-core
+secretRef:
+  name: secret1
+# namespace: namespace-other-than-'garden-core' // optional
+quotas: []
+# - name: quota-1
+# # namespace: namespace-other-than-'garden-core' // optional
+```
+
+### `Quota` resource
+
+Special note: No modifications needed compared to the current `garden.sapcloud.io/v1beta1.Quota` resource.
+
+```yaml
+apiVersion: core.gardener.cloud/v1alpha1
+kind: Quota
+metadata:
+  name: trial-quota
+  namespace: garden-trial
+spec:
+  scope:
+    apiGroup: core.gardener.cloud
+    kind: Project
+# clusterLifetimeDays: 14
+  metrics:
+    cpu: "200"
+    gpu: "20"
+    memory: 4000Gi
+    storage.standard: 8000Gi
+    storage.premium: 2000Gi
+    loadbalancer: "100"
+```
+
+### `BackupBucket` resource
+
+Special note: This new resource is cluster-scoped.
+
+```yaml
+# See also: https://github.com/gardener/gardener/blob/master/docs/proposals/02-backupinfra.md.
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: backup-operator-provider
+  namespace: backup-garden
+type: Opaque
+data:
+  # <some-provider-specific data keys>
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-alicloud/example/30-backupbucket.yaml#L9-L11
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-azure/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-gcp/example/30-backupbucket.yaml#L9
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-openstack/example/30-backupbucket.yaml#L9-L13
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: BackupBucket
+metadata:
+  name: providerprofile1-random[:5]
+  ownerReferences:
+  - kind: ProviderProfile
+    name: providerprofile1
+spec:
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    region: europe-central-1
+  seed: seed1
+  secretRef:
+    name: backup-operator-provider
+    namespace: backup-garden
+status:
+  lastOperation:
+    description: Backup bucket has been successfully reconciled.
+    lastUpdateTime: '2020-04-13T14:34:27Z'
+    progress: 100
+    state: Succeeded
+    type: Reconcile
+  observedGeneration: 1
+```
+
+### `BackupEntry` resource
+
+Special note: This new resource is cluster-scoped.
```

Comment:
I'll probably rename this resource to `ClusterBackupEntry` to avoid confusion.

Reply:
@swapnilgm any thoughts?
I have no strong opinion on this, IMHO it makes only sense to prefix it with `Cluster` if we plan to introduce a namespaced `BackupEntry`. I don't think we do.

Reply:
Currently BackupEntry is namespaced. ANd there is no plan/need to introduce cluster scoped BackupEntry. So, i don't think there will be any confusion.